### PR TITLE
OHRM5X-1394:Functional Automation - Job Category

### PR DESCRIPTION
**Pull request by OvaraPerera**

### CODE REVIEW

File: src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
+describe('Admin - Job Category', function () {
+  beforeEach(function () {
+    cy.task('db:reset');
+    cy.viewport(1280, 720);
```
Comment (RajithaKumara) [inline src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js:4]:
```suggestion
    cy.fixture('viewport').then(({HD}) => {
      cy.viewport(HD.width, HD.height);
    });
```
@Super-Chama is this correct?

File: src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
+describe('Admin - Job Category', function () {
```
Comment (RajithaKumara) [inline src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js:1]: missing copyright header

File: src/test/functional/cypress/support/oxd.js [hunk 3]
```diff
@@ -45,6 +45,7 @@ const OXD_ELEMENTS = {
   inputGroup: '.oxd-input-group',
   pageContext: '.oxd-layout-context',
   pageTitle: '.orangehrm-main-title',
+  numRecords: '.oxd-text--span',
```
Comment (RajithaKumara) [inline src/test/functional/cypress/support/oxd.js:48]: @Super-Chama is this selector enough?

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js (hunk 1 above)
Comment (Super-Chama) [reply src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js:4]: yeah this should be the way

File: src/test/functional/cypress/support/oxd.js (hunk 3 above)
Comment (Super-Chama) [reply src/test/functional/cypress/support/oxd.js:48]: seems not, i'll have to check this one

Comment (Super-Chama) [reply src/test/functional/cypress/support/oxd.js:48]: there are no classes or identifier for this. I will keep this noted. I may have to make a new PR and refactor these with more custom methods

Comment (Super-Chama) [reply src/test/functional/cypress/support/oxd.js:48]: there are no classes or identifier for this. I will keep this noted. I may have to make a new PR and refactor these with more custom methods

File: src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  //Update
+  describe('Update job Category', function () {
+    it('Edit job category', function () {
+      cy.loginTo(this.user, '/admin/saveJobCategory/1');
+      cy.getOXD('form').within(() => {
+        cy.getOXDInput('Name').clear().type(this.strings.chars50.text);
+        cy.getOXD('button').contains('Save').click();
+      });
+      cy.wait('@updateJobCategories');
+      cy.toast('success', 'Successfully Updated');
```
Comment (RajithaKumara) [inline src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js:93]: Don't we need to visit back to `/admin/saveJobCategory/1` and see actually name changed?

File: src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    });
+  });
+  //Delete
```
Comment (RajithaKumara) [inline src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js:96]:
```suggestion

  //Delete
```
keep new line between describe blocks and test blocks, it's readable then

File: src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  describe('Delete job Category', function () {
+    it('Delete a single job category', function () {
+      cy.loginTo(this.user, '/admin/jobCategory');
+      cy.wait('@getJobCategories');
+      cy.get(
+        ':nth-child(1) > .oxd-table-row > [style="flex-shrink: 1;"] > .oxd-table-cell-actions > :nth-child(1)',
+      ).click();
```
Comment (RajithaKumara) [inline src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js:103]: @Super-Chama I think may need help here

File: src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+      cy.getOXD('button').contains('Yes, Delete').click();
+      cy.wait('@getJobCategories');
+      cy.toast('success', 'Successfully Deleted');
+    });
+    it('Bulk Delete job categories', function () {
+      cy.loginTo(this.user, '/admin/jobCategory');
+      cy.wait('@getJobCategories');
+      cy.get(
+        '.oxd-table-header > .oxd-table-row > :nth-child(1) > .oxd-checkbox-wrapper > label > .oxd-checkbox-input > .oxd-icon',
+      ).click();
```
Comment (RajithaKumara) [inline src/test/functional/cypress/integration/feature/admin/jobCategories.spec.js:113]: *

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**